src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -7995,6 +7995,11 @@ private void ParseDeclarationModifiers(SyntaxListBuilder list)
                 {
                     mod = this.AddError(mod, ErrorCode.ERR_BadMemberFlag, mod.Text);
                 }
+                else if (list.Any(mod.Kind))
+                {
+                    // check for duplicates, can only be const
+                    mod = this.AddError(mod, ErrorCode.ERR_TypeExpected, mod.Text);
```

Comment:
@AlekseyTs: Agreed, but I for now I just recreated the behavior of fields. The previous compiler reports a different error if I recall correctly, but for consistency I chose Roslyn's behavior.